pkg/db/db.go

```diff
@@ -41,71 +48,54 @@ func Provision(filename string, dbCon *gorm.DB) error {
 	}
 	defer file.Close()
 
-	resources := ProvisionedResources{}
+	resources := ProvisionedResources{} //nolint:exhaustruct
 
 	err = yaml.NewDecoder(file).Decode(&resources)
 	if err != nil {
 		return fmt.Errorf("error decoding provisioning file `%s`: %w", filename, err)
 	}
 
-	for _, component := range resources.Components {
-		err = dbCon.Save(component).Error
-		if err != nil {
-			return fmt.Errorf("error saving component `%s`: %w", component.DisplayName, err)
-		}
-	}
+	initialPhaseGeneration := 1
 
-	for _, impactType := range resources.ImpactTypes {
-		err = dbCon.Save(impactType).Error
-		if err != nil {
-			return fmt.Errorf("error saving impact type `%s`: %w", impactType.Slug, err)
-		}
-	}
+	// check if already provisioned
+	var lastPhase Phase
 
-	var phaseOrder uint
-	for phaseIndex := range resources.Phases {
-		resources.Phases[phaseIndex].Order = phaseOrder
+	res := dbCon.
```

Comment:
This `Provision` function does not seem to use a database transaction, enabling (1) inconsistent states if some `Create` call fails and (2) race conditions, if there is more than one server replica running.